packages/framework-provider-aws/src/library/events-adapter.ts

```diff
@@ -88,7 +88,37 @@ export async function storeEvents(
   config: BoosterConfig,
   logger: Logger
 ): Promise<void> {
-  // TODO: Manage query pagination and db.batchWrite limit of 25 operations at a time
+  const batchSize = 25
+  let processedItems = 0
+  let currentBatch = getBatchFromEnvelopes(eventEnvelopes, processedItems, batchSize)
+  while (currentBatch.length != 0) {
+    await persistBatch(logger, currentBatch, config, dynamoDB)
+    processedItems += currentBatch.length
+    currentBatch = getBatchFromEnvelopes(eventEnvelopes, processedItems, batchSize)
+  }
```

Comment:
This is one of those rare scenarios where a `do..while` could make sense:

```suggestion
  do {
    currentBatch = getBatchFromEnvelopes(eventEnvelopes, processedItems, batchSize)
    await persistBatch(logger, currentBatch, config, dynamoDB)
    processedItems += currentBatch.length
  } while (currentBatch.length > 0)
```

I'm also paranoid with strict comparations with indexes, so I'd use a `>` instead of `!=`

Reply:
Agree here